Eliminated additional data movement in RRTMGP

johnmauff opened the pull request:
This is a follow-up for PR #24.  It reduces additional data movement in the rte_sw() and rte_lw() subroutines.  It also eliminates data movement within the calls to gas_optics() by moving the data movement outside of the subroutine call tree.

johnmauff committed:
Eliminated additional data movement in RRTMGP

gdicker1 reviewed (commented):
Thanks @johnmauff this looks good.

Would you mind if I also add an update to the Externals_CAM.cfg to bring in the [EWOrg/rte-rrtmgp #1](https://github.com/EarthWorksOrg/rte-rrtmgp/pull/1) changes?

Commit:
Bring in changes from EarthWorksOrg/rte-rrtmgp PR#1

These changes include removing a (now unneeded) temp variable and some
unnecessary OpenACC copyout directives.

johnmauff commented:
Dylan,

   Yes, that would be a good idea.  The changes to radiation_tend() depend
on the use of the rte-rrtmgp PR.

John

On Thu, Aug 1, 2024 at 1:11 PM G. Dylan Dickerson ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Thanks @johnmauff <https://github.com/johnmauff> this looks good.
>
> Would you mind if I also add an update to the Externals_CAM.cfg to bring
> in the EWOrg/rte-rrtmgp #1
> <https://github.com/EarthWorksOrg/rte-rrtmgp/pull/1> changes?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/EarthWorksOrg/CAM/pull/25#pullrequestreview-2213693001>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADH7NUTU73IT3DW67ZQCXGDZPKB7LAVCNFSM6AAAAABLZNTYG2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEMJTGY4TGMBQGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

gdicker1 commented:
Force push [5b267b8 ](https://github.com/EarthWorksOrg/CAM/commit/5b267b8b16405b24bd658f49e6d53cc729ea8728) to [044002f](https://github.com/EarthWorksOrg/CAM/commit/044002fb8fb843f16d8195c067491fea7e840cef) because I couldn't leave a commit message typo alone. Otherwise, only change made was to update Externals_CAM.cfg

gdicker1 reviewed (approved):
These changes look good to me now. I plan to merge this in about an hour
